go-controller/pkg/ovn/acl.go

```diff
@@ -50,15 +54,43 @@ func policyTypeToAclType(policyType knet.PolicyType) aclType {
 	}
 }
 
+func policyTypeStrToAclType(policyType string) aclType {
+	switch policyType {
+	case string(knet.PolicyTypeEgress):
+		return lportEgressAfterLB
+	case string(knet.PolicyTypeIngress):
+		return lportIngress
+	default:
+		panic(fmt.Sprintf("Failed to convert PolicyType to aclType: unknown policyType type %s", policyType))
+	}
+}
+
 // hash the provided input to make it a valid portGroup name.
 func hashedPortGroup(s string) string {
 	return util.HashForOVN(s)
 }
 
+// acl.Name is cropped to 64 symbols and is used for logging.
+// this function will create the name based on DbObjectIDs, replacing all owner info with just the 2-letter
+// type of owner object.
+func getACLName(dbIDs *libovsdbops.DbObjectIDs) string {
+	t := dbIDs.GetIDsType()
+	aclPrefix := ""
+	switch {
+	case t.IsSameType(libovsdbops.ACLEgressFirewall):
+		aclPrefix = "EF"
+	case t.IsSameType(libovsdbops.ACLNetpolNamespace) || t.IsSameType(libovsdbops.ACLNetworkPolicy) || t.IsSameType(libovsdbops.ACLNetpolNode):
+		aclPrefix = "NP"
+	case t.IsSameType(libovsdbops.ACLMulticastGlobal) || t.IsSameType(libovsdbops.ACLMulticastNamespace):
+		aclPrefix = "MC"
+	}
```

Comment:
missing this panic still?

Reply:
I don't know exactly by who or how is this log being consumed. But assuming it's an end user that at the end has to understand what is going on, could we just use something more simple and less cryptic like:

`Network Policy <name>/<namespace>`
and
`Egress Firewall <namespace>`

As long as in the source (in the network policy or egress firewall) things within are unnamed, we shouldn't necessarily be responsible to give them a name for the user to be able to correlate them within the source.

Reply:
yeah it is user facing https://docs.openshift.com/container-platform/4.12/networking/ovn_kubernetes_network_provider/logging-network-policy.html and may be used e.g. for debugging. The problem is that only 64 symbols are allowed, and just namespace name can be that long (and we are getting more automatically created namespaces with long names). So the point of this function was to provide as much info as possible in case the name gets cropped.
So I made acronym feature names, to save some namespace name symbols. Indexes may also be useful to see which exact gress/firewall rule affects traffic. does it make sense @jcaamano ?

Reply:
LOL the way these are consumed is not that great seeing the docs. 

Like I said, I wouldn't consider the fact that names can be cropped our problem anymore as long as we don't depend on it. I don't think we need to specify details like indexes or such either. Unless we do have a use of it when troubleshooting but I guess we shouldn't actually (since it can be cropped).

I am just saying that we shouldn't be worrying too much about it now and it just kinda relegates this to be user facing only and I would focus on helping the user understand what's going on. Not a strong opinion though. We could always clarify on the documentation what these mean.

Reply:
then I guess the expectation of this function would be that someone only calls it if they are using the right type? I would have a default case that logs someone is trying to use this function for an unknown type or add comments to the function being explicit about what this function is used for. Maybe change the name of the function for getACLNameForLogging or something.

Reply:
not really, this function is called from `BuildACL` for every type of acl, and it fills the name only for acls that need it (the ones that are logged). getACLName comment should explain that, maybe I can update it somehow to make more obvious?